openmetadata-ui/src/main/resources/ui/src/components/ContainerDetail/ContainerDataModel/ContainerDataModel.tsx

```diff
@@ -0,0 +1,284 @@
+      setTagList(tagsAndTerms);
+    } catch (error) {
+      setTagList([]);
+      setTagFetchFailed(true);
+    } finally {
+      setIsTagLoading(false);
+    }
+  };
+
+  const handleFieldTagsChange = async (selectedTags: EntityTags[] = []) => {
+    if (!isUndefined(editContainerColumnTags)) {
+      const newSelectedTags: TagOption[] = selectedTags.map((tag) => ({
+        fqn: tag.tagFQN,
+        source: tag.source,
+      }));
+
+      const containerDataModel = cloneDeep(dataModel);
+
+      updateContainerColumnTags(
+        containerDataModel?.columns,
+        editContainerColumnTags?.name,
+        newSelectedTags
+      );
+
+      await onUpdate(containerDataModel);
+    }
+    setEditContainerColumnTags(undefined);
+  };
+
+  const handleAddTagClick = (record: Column) => {
+    if (isUndefined(editContainerColumnTags)) {
+      setEditContainerColumnTags(record);
+      // Fetch tags and terms only once
+      if (tagList.length === 0 || tagFetchFailed) {
+        fetchTags();
+      }
+    }
+  };
+
+  const handleContainerColumnDescriptionChange = async (
+    updatedDescription: string
+  ) => {
+    if (!isUndefined(editContainerColumnDescription)) {
+      const containerDataModel = cloneDeep(dataModel);
+      updateContainerColumnDescription(
+        containerDataModel?.columns,
+        editContainerColumnDescription?.name,
+        updatedDescription
+      );
+      await onUpdate(containerDataModel);
+    }
+    setEditContainerColumnDescription(undefined);
+  };
+
+  const renderContainerColumnDescription: CellRendered<Column, 'description'> =
+    (description, record, index) => {
+      return (
+        <Space
+          className="custom-group w-full"
+          data-testid="description"
+          id={`field-description-${index}`}
+          size={4}>
+          <>
+            {description ? (
+              <RichTextEditorPreviewer markdown={description} />
+            ) : (
+              <Typography.Text className="tw-no-description">
+                {t('label.no-entity', {
+                  entity: t('label.description'),
+                })}
+              </Typography.Text>
+            )}
+          </>
+          {isReadOnly && !hasDescriptionEditAccess ? null : (
+            <Button
+              className="p-0 opacity-0 group-hover-opacity-100"
+              data-testid="edit-button"
+              icon={<EditIcon width="16px" />}
+              type="text"
+              onClick={() => setEditContainerColumnDescription(record)}
+            />
+          )}
+        </Space>
+      );
+    };
+
+  const renderContainerColumnTags: CellRendered<Column, 'tags'> = (
+    tags,
+    record: Column
+  ) => {
+    const isSelectedField = editContainerColumnTags?.name === record.name;
+    const styleFlag = isSelectedField || !isEmpty(tags);
+
+    return (
+      <>
+        {isReadOnly ? (
+          <Space wrap>
+            <TagsViewer sizeCap={-1} tags={tags || []} />
+          </Space>
```

Comment:
I feel we should add this within <TagsViewer /> component instead parent

Reply:
Thanks, I will check.

Reply:
@chirag-madlani, I have checked this and we have multiple references for <TagsViewer so I will take care of this in my next patch.